shared/src/main/scala/effekt/Typer.scala

```diff
@@ -115,7 +115,9 @@ class Typer extends Phase[Module, Module] { typer =>
               val effectOp = d.definition
               val bt = Context.blockTypeOf(effectOp)
               val ps = checkAgainstDeclaration(op.name, bt.params, params)
-              val resumeType = BlockType(Nil, List(List(effectOp.ret.get.tpe)), ret / Pure)
+              val effs = effectOp.ret.get.effects - effectOp.effect
+              val resumeArgType = BlockType(Nil, List(Nil), Effectful(effectOp.ret.get.tpe, effs))
+              val resumeType = BlockType(Nil, List(List(resumeArgType)), ret / Pure)
```

Comment:
```suggestion
              val BlockType(_, pms, tpe / effs) = Context.blockTypeOf(effectOp)
              val ps = checkAgainstDeclaration(op.name, pms, params)
 
              val resumeType = if (effectOp.isBidirectional) {
                // resume { e }
                BlockType(Nil, List(List(BlockType(Nil, List(Nil), tpe / effectOp.otherEffects))), ret / Pure)
              } else {
                // resume(v)
                BlockType(Nil, List(List(tpe)), ret / Pure)
              }
```

We should not allow `resume(e)` syntax for bidirectional effects. Otherwise this will lead to a runtime error "f.apply is not a function".

The methods `isBidirectional` and `otherEffects` should be added to the EffectOp symbol, like you did previously:

```
def otherEffects: Effects = ret.get.effects - effect
def isBidirectional: Boolean = otherEffects.nonEmpty
```